android/java/org/chromium/chrome/browser/crypto_wallet/util/Utils.java

```diff
@@ -540,6 +540,28 @@ public static String toWeiHex(String number) {
         return "0x" + bigNumber.toString(16);
     }
 
+    public static String toHex(String number) {
+        if (number.isEmpty()) {
+            return "0x0";
+        }
+        BigInteger bigNumber = new BigInteger(number, 10);
```

Comment:
let's catch a `NumberFormatException` here and below in case String contains an invalid number https://docs.oracle.com/javase/7/docs/api/java/math/BigInteger.html#BigInteger(java.lang.String,%20int)

Reply:
yes though I think it should be the front end/client to apply the validation or handle the errors only if required, then it will be easy to find unexpected values sooner than later. WDYT?

Reply:
yeah, could be done on a higher level.